homeassistant/components/sensor/filter.py

```diff
@@ -0,0 +1,177 @@
+"""
+Support for filtering for sensor values.
+
+Example configuration:
+
+sensor:
+  - platform: simulated
+    name: 'simulated relative humidity'
+    unit: '%'
+    amplitude: 0 # Turns off sine wave
+    mean: 50
+    spread: 10
+    seed: 999
+
+  - platform: filter
+    entity_id: sensor.simulated_relative_humidity
+    name: lowpass
+    options:
+      time_constant: 10
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filter/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_ENTITY_ID, ATTR_UNIT_OF_MEASUREMENT, STATE_UNKNOWN)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.filter import (
+    Filter, FILTERS)
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PRE_FILTER = 'pre_filter_state'
+
+CONF_FILTER_OPTIONS = 'options'
+CONF_FILTER_NAME = 'filter'
+
+DEFAULT_NAME_TEMPLATE = "{} filter {}"
+ICON = 'mdi: chart-line-variant'
+
+# ALL filter arguments must be OPTIONAL
+FILTER_SCHEMA = vol.Schema({
+    vol.Optional('time_constant'): vol.Coerce(int),
+    vol.Optional('constant'): vol.Coerce(float)
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Required(CONF_FILTER_NAME):
+        vol.Any(*FILTERS),
```

Comment:
Use `vol.In(list(FILTERS))`

Reply:
find my version quite pythonic :) but changing...